app/controllers/case_court_reports_controller.rb

```diff
@@ -14,15 +14,16 @@ def index
   # GET /case_court_reports/:id
   def show
     authorize CaseCourtReport
-    unless @casa_case
+    if !@casa_case || !@casa_case.court_report.attached?
       flash[:alert] = "Report #{params[:id]} is not found."
       redirect_to(case_court_reports_path) and return # rubocop:disable Style/AndOr
     end
 
     respond_to do |format|
       format.docx do
```

Comment:
Moving the court report generation in line 24 to the `generate` method. 

Right now, clicking the `Click to download` button triggers the report generation, but I think we'd actually want the court report to be generated (and saved to `casa_case`) on `Generate report`.